core/capabilities/gateway_connector/service_wrapper_test.go

```diff
@@ -0,0 +1,66 @@
+package gatewayconnector
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	chainlink "github.com/smartcontractkit/chainlink/v2/core/services/chainlink/config"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
+	ksmocks "github.com/smartcontractkit/chainlink/v2/core/services/keystore/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/test-go/testify/mock"
+)
+
+// Unit test that creates the ServiceWrapper object and then calls Start() can Close() on it.
+// Take inspiration from functions/plugin_test.go and functions/connector_handler_test.go on how to mock the dependencies.
+//
+// Test valid NodeAddress and an invalid one (i.e. key doesn't exit).
+
+func TestGatewayConnectorServiceWrapper(t *testing.T) {
+	t.Parallel()
+
+	logger := logger.TestLogger(t)
+	_, addr := testutils.NewPrivateKeyAndAddress(t)
+
+	config, err := chainlink.GeneralConfigOpts{}.New()
+	// I don't think this is the right way to get the GatewayConnector in.
```

Comment:
It's not. You should do it in the line above. Instead of creating an empty GeneralConfigOpts{} object, you create one with desired config already in it. Something like:
```
config, err := GeneralConfigOpts{
	Config: Config{
		Core: toml.Core{
			Capabilities: toml.Capabilities{
				GatewayConnector: toml.GatewayConnector{
					// all the fields here
				},
			},
		},
	},
}.New()
```